src/lib/storage/lz4_segment.hpp

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <boost/hana/contains.hpp>
+#include <boost/hana/tuple.hpp>
+#include <boost/hana/type.hpp>
+
+#include <type_traits>
+
+#include <array>
+#include <memory>
+
+#include "base_encoded_segment.hpp"
+#include "storage/vector_compression/base_compressed_vector.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+class BaseCompressedVector;
+
+template <typename T>
+class LZ4Segment : public BaseEncodedSegment {
+ public:
+  /*
+   * An LZ4 segment is a container for an LZ4 compressed segment. It contains the compressed data, the necessary
+   * metadata and the ability to decompress the data again.
+   *
+   * @param compressed_data The char vector that contains the LZ4 compressed segment data as binary blob.
+   * @param null_values Boolean vector that contains the information which row is null and which is not null.
+   * @param offsets If this segment is not a std::string segment this will be a null pointer. Otherwise it contains
+   *                the offsets for the compressed strings. The offset at position 0 is the character index of the
+   *                string at index 0. Its (exclusive) end is at the offset at position 1. The last string ends at the
+   *                end of the compressed data (since there is offset after it that specifies the end offset).
+   * @param compressed_size The size of the compressed data vector (the return value of LZ4)
+   * @param decompressed_size The size in bytes of the decompressed data vector.
+   * @param num_elements The number of rows that are compressed in this segment.
```

Comment:
Why do I need this if I have null_values.size?

Reply:
That's true. We'll remove it.